standard/PortalStatistics.lua

```diff
@@ -0,0 +1,1140 @@
+			.. 'opponentplayers, opponentname, opponenttype',
+		conditions = conditions:toString(),
+		limit = 1000,
+	}
+
+	local function returnOpponent(item)
+		local opponentNames = {}
+		if args.opponentType == Opponent.solo then
+			for playerIndex = 1,10 do
+				local name = string.gsub(item.opponentplayers['p' .. playerIndex] or '', ' ', '_')
+				table.insert(opponentNames, name)
+			end
+		elseif args.opponentType == Opponent.team and item.opponenttype == Opponent.team then
+			local name = string.gsub(item.opponentname or '', ' ', '_')
+			table.insert(opponentNames, name)
+		end
+		return opponentNames
+	end
+
+	local processData = function(item)
+		local placement = string.sub(item.placement, 1, 1)
+		for _, opponent in pairs(returnOpponent(item) or {}) do
+			if not data[opponent] then
+				data[opponent] = {['1'] = 0, ['2'] =  0, ['3'] = 0, showWins = 0, sWinData = {}}
+			end
+			if placement == FIRST and item.liquipediatier == STIER and item.liquipediatiertype ~= SHOWMATCH then
+				table.insert(data[opponent].sWinData, {
+						icon = item.icon,
+						iconDark = item.icondark,
+						pagename = item.pagename,
+						shortname = item.shortname
+					}
+				)
+			end
+			if placement == FIRST and item.liquipediatiertype == SHOWMATCH then
+				data[opponent].showWins = data[opponent].showWins + 1
+			elseif item.liquipediatiertype ~= SHOWMATCH then
+				data[opponent][placement] = data[opponent][placement] + 1
+			end
+		end
+	end
+
+	Lpdb.executeMassQuery('placement', queryParameters, processData)
+
+	return data
+end
+
+
+--[[
+Section: Display Functions
+]]--
+
+
+function StatisticsPortal._earningsTableHeader(args)
+	local columnText = args.opponentType == Opponent.team and 'Organization' or 'Player'
+
+	local row = mw.html.create('tr')
+		:tag('th'):wikitext('#'):addClass('unsortable'):done()
+		:tag('th'):wikitext(columnText):addClass('unsortable'):done()
+		:tag('th'):wikitext('Achievements'):css('width', '200px'):addClass('unsortable'):done()
+		:tag('th'):wikitext(Medal['1']):done()
+		:tag('th'):wikitext(Medal['2']):done()
+		:tag('th'):wikitext(Medal['3']):done()
+
+	if Logic.readBool(args.displayShowMatches) then
+		row:tag('th'):wikitext('Show<br>Match'):done()
+	end
+
+	row:tag('th')
+		:tag('abbr')
+		:attr('title', 'Total earnings across all games')
+		:wikitext('Earnings'):done()
+
+	return row
+end
+
+
+function StatisticsPortal._earningsTableRow(args, placements, earnings, opponentIndex, opponentDisplay)
+	local row = mw.html.create('tr')
+		:css('line-height', '25px')
+		:css('text-align', 'center')
+		:tag('td'):wikitext(opponentIndex):done()
+		:tag('td'):css('text-align', 'left'):node(opponentDisplay):done()
+		:tag('td'):wikitext(StatisticsPortal._achievementsDisplay(placements.sWinData or {})):done()
+		:tag('td'):wikitext(placements['1'] or '0'):done()
+		:tag('td'):wikitext(placements['2'] or '0'):done()
+		:tag('td'):wikitext(placements['3'] or '0'):done()
```

Comment:
```suggestion
		:tag('td'):wikitext(placements['3'] or '0')
```

Reply:
Just to understand, this is because it's the last one, and doesn't need :done() to close it like the others do?

Reply:
yeah the done is only needed if you want to attach stuff on the next upper level (i.e. here the tr)
since the next tag is directly attached to the tr element again no need for the done